fetch.bs

```diff
@@ -1673,53 +1682,67 @@ is to return the result of <a>serializing a request origin</a> with <var>request
 
   <p class="note no-backref">This operation cannot throw an exception.
 
- <li><p>Let <var>read</var> be the result of <a lt="read a chunk" for=ReadableStream>reading a
- chunk</a> from <var>body</var>'s <a for=body>stream</a> with <var>reader</var>.
+ <li><p>Perform the <a>transmit-body loop</a> given <var>request</var>, <var>body</var>, and
+ <var>reader</var>.
+</ol>
 
- <li>
-  <p><a>In parallel</a>, while true:
+<p>To perform the <dfn>transmit-body loop</dfn> given <var>request</var>, <var>body</var>, and
+<var>reader</var>:
 
-  <ol>
-   <li>
-    <p>Run these steps, but <a>abort when</a> the ongoing fetch is <a for=fetch>terminated</a>:
+<ol>
+ <li>
+  <p>Let <var>readRequest</var> be the following <a>read request</a>:
 
+  <dl>
+   <dt><a for="read request">chunk steps</a>, given <var>chunk</var>
+   <dd>
     <ol>
-     <li><p>Wait for <var>read</var> to be fulfilled or rejected.
+     <li><p>If the ongoing fetch is <a for=fetch>terminated</a>, then abort these steps.
+
+     <li><p>If <var>chunk</var> is not a {{Uint8Array}} object,
+     <a lt=terminated for=fetch>terminate</a> the ongoing fetch and abort these steps.
+
+     <li><p>Let <var>bs</var> be the <a>byte sequence</a> represented by the {{Uint8Array}} object.
 
      <li>
-      <p>If <var>read</var> is fulfilled with an object whose <code>done</code> property is false
-      and whose <code>value</code> property is a <code>Uint8Array</code> object, then run these
-      steps:
+      <p><a>In parallel</a>:
 
       <ol>
-       <li><p>Let <var>bs</var> be the <a>byte sequence</a> represented by the
-       <code>Uint8Array</code> object.
+       <li><p>Transmit <var>bs</var>. Whenever one or more bytes are transmitted, increase
+       <var>body</var>'s <a for=body>transmitted bytes</a> by the number of transmitted bytes and
+       <a>queue a fetch task</a> on <var>request</var> to <a>process request body</a>
+       for <var>request</var>.
 
-       <li>
-        <p>Transmit <var>bs</var>. Whenever one or more bytes are transmitted, increase
-        <var>body</var>'s <a for=body>transmitted bytes</a> by the number of transmitted bytes and
-        <a>queue a fetch task</a> on <var>request</var> to <a>process request body</a>
-        for <var>request</var>.
+       <p class="note no-backref">This step blocks until <var>bs</var> is fully transmitted.
 
-        <p class="note no-backref">This step blocks until <var>bs</var> is fully transmitted.
+      <li><p>If the ongoing fetch is not <a for=fetch>terminated</a>, then perform the
+      <a>transmit-body loop</a> given <var>request</var>, <var>body</var>, and <var>reader</var>.
```

Comment:
This seems like cheating. Don't we have to post a task to get back out of "in parallel"?

Reply:
Yes, oops, I meant to do that.